src/sass_types.h

```diff
@@ -0,0 +1,225 @@
+        proto->Set(NanNew("getA"), FunctionTemplate::New(GetA)->GetFunction());
+        proto->Set(NanNew("setR"), FunctionTemplate::New(SetR)->GetFunction());
+        proto->Set(NanNew("setG"), FunctionTemplate::New(SetG)->GetFunction());
+        proto->Set(NanNew("setB"), FunctionTemplate::New(SetB)->GetFunction());
+        proto->Set(NanNew("setA"), FunctionTemplate::New(SetA)->GetFunction());
+      }
+
+      static NAN_METHOD(GetR);
+      static NAN_METHOD(GetG);
+      static NAN_METHOD(GetB);
+      static NAN_METHOD(GetA);
+      static NAN_METHOD(SetR);
+      static NAN_METHOD(SetG);
+      static NAN_METHOD(SetB);
+      static NAN_METHOD(SetA);
+  };
+
+
+  class Boolean : public CoreValue<Boolean> {
+    public:
+      Boolean(Sass_Value* v) : CoreValue(v) {}
+      static char const* get_constructor_name() { return "SassBoolean"; }
+      static Sass_Value* construct(const Arguments&);
+
+      static void initPrototype(Handle<ObjectTemplate> proto) {
+        proto->Set(NanNew("getValue"), FunctionTemplate::New(GetValue)->GetFunction());
+        proto->Set(NanNew("setValue"), FunctionTemplate::New(SetValue)->GetFunction());
+      }
+
+      static NAN_METHOD(GetValue);
+      static NAN_METHOD(SetValue);
+  };
+
+
+  class List : public CoreValue<List> {
+    public:
+      List(Sass_Value* v) : CoreValue(v) {}
+      static char const* get_constructor_name() { return "SassList"; }
+      static Sass_Value* construct(const Arguments&);
+
+      static void initPrototype(Handle<ObjectTemplate> proto) {
+        proto->Set(NanNew("getLength"), FunctionTemplate::New(GetLength)->GetFunction());
+        proto->Set(NanNew("getSeparator"), FunctionTemplate::New(GetSeparator)->GetFunction());
+        proto->Set(NanNew("setSeparator"), FunctionTemplate::New(SetSeparator)->GetFunction());
+        proto->Set(NanNew("getValue"), FunctionTemplate::New(GetValue)->GetFunction());
+        proto->Set(NanNew("setValue"), FunctionTemplate::New(SetValue)->GetFunction());
+      }
+
+      static NAN_METHOD(GetValue);
+      static NAN_METHOD(SetValue);
+      static NAN_METHOD(GetSeparator);
+      static NAN_METHOD(SetSeparator);
+      static NAN_METHOD(GetLength);
+  };
+
+
+  class Map : public CoreValue<Map> {
+    public:
+      Map(Sass_Value* v) : CoreValue(v) {}
+      static char const* get_constructor_name() { return "SassMap"; }
+      static Sass_Value* construct(const Arguments&);
+
+      static void initPrototype(Handle<ObjectTemplate> proto) {
+        proto->Set(NanNew("getLength"), FunctionTemplate::New(GetLength)->GetFunction());
+        proto->Set(NanNew("getKey"), FunctionTemplate::New(GetKey)->GetFunction());
+        proto->Set(NanNew("setKey"), FunctionTemplate::New(SetKey)->GetFunction());
+        proto->Set(NanNew("getValue"), FunctionTemplate::New(GetValue)->GetFunction());
+        proto->Set(NanNew("setValue"), FunctionTemplate::New(SetValue)->GetFunction());
+      }
```

Comment:
You have already made the code much readable than it was before. IMO with this, it will make the code discoverability much easy for future contributors.

My vote is for even making code/implementation files separate for each derived class. :)